src/test/java/org/dmfs/oauth2/client/http/entities/XWwwFormUrlEncodedEntityTest.java

```diff
@@ -121,40 +99,27 @@ public void testWriteToOutputStream() throws IOException
     @Test
     public void testUrlEncodingWithAFewSpecialCharacters()
     {
+
+        ParameterType<CharSequence> keyWithSpaces = new BasicParameterType<>("key and space", TextValueType.INSTANCE);
         assertEquals("key+and+space=value+with+spaces",
-                new XWwwFormUrlEncodedEntity(new ImmutableEntry("key and space", "value with spaces")).toString());
+                new XWwwFormUrlEncodedEntity(keyWithSpaces.parameter("value with spaces")).toString());
 
+        ParameterType<CharSequence> keyWithQuotes = new BasicParameterType<>("key\"doublequote", TextValueType.INSTANCE);
         assertEquals("key%22doublequote=",
-                new XWwwFormUrlEncodedEntity(new ImmutableEntry("key\"doublequote", "")).toString());
+                new XWwwFormUrlEncodedEntity(keyWithQuotes.parameter("")).toString());
 
+        ParameterType<CharSequence> keyaa = new BasicParameterType<>("aa", TextValueType.INSTANCE);
         assertEquals("aa=%7B",
-                new XWwwFormUrlEncodedEntity(new ImmutableEntry("aa", "{")).toString());
+                new XWwwFormUrlEncodedEntity(keyaa.parameter("{")).toString());
 
         // equals ('=')
+        ParameterType<CharSequence> keyWithEquals = new BasicParameterType<>("key=with=equals", TextValueType.INSTANCE);
         assertEquals("key%3Dwith%3Dequals=value%3Dwith%3Dequals",
-                new XWwwFormUrlEncodedEntity(new ImmutableEntry("key=with=equals", "value=with=equals")).toString());
+                new XWwwFormUrlEncodedEntity(keyWithEquals.parameter("value=with=equals")).toString());
 
         // and ('&')
+        ParameterType<CharSequence> keyWithAmpersand = new BasicParameterType<>("key&with&and", TextValueType.INSTANCE);
         assertEquals("key%26with%26and=value%26with%26and",
-                new XWwwFormUrlEncodedEntity(new ImmutableEntry("key&with&and", "value&with&and")).toString());
+                new XWwwFormUrlEncodedEntity(keyWithAmpersand.parameter("value&with&and")).toString());
     }
-
-
-    @Test
-    public void testIgnoreEntries_whenEntryIsNull_orKeyIsNullOrEmpty_orValueIsNull()
```

Comment:
`null` parameters are no longer allowed. This is supposed to throw a NPE when you pass `null`.
Since `null` is not a valid input we don't need to test it explicitly.

Also, `null` values and keys are handled by the parameters themselves so this is no longer subject to `XWwwFormUrlEncodedEntity`.